packages/yew/src/functional/mod.rs

```diff
@@ -65,125 +67,85 @@ pub use yew_macro::hook;
 type ReRender = Rc<dyn Fn()>;
 
 /// Primitives of a Hook state.
-pub(crate) trait Effect {
-    fn rendered(&self) {}
+pub(crate) trait Effect: ToAny {
+    fn rendered(&mut self) {}
+}
+
+pub(crate) trait ToAny: Any {
+    fn as_any(&mut self) -> &mut dyn Any;
+}
+
+impl<T: Effect> ToAny for T {
+    fn as_any(&mut self) -> &mut dyn Any {
+        self
+    }
 }
```

Comment:
This looks a bit messy to me. Perhaps we could merge `ToAny` into `Effect` with a default implementation?

Reply:
This exists to work around object safety. Note that `Effect: Sized` would not be object-safe but the blanket impl only works for `T: Sized` (otherwise the coercion is illegal).